app/src/androidTest/java/ch/epfl/sdp/MarketEspresso/NewMarketActivityTest.java

```diff
@@ -0,0 +1,167 @@
+                    "android.permission.ACCESS_FINE_LOCATION");
+
+    private void clickOnItem(int buttonId, int cardId, int position){
+        ViewInteraction appCompatImageButton = onView(
+                allOf(withId(buttonId),
+                        childAtPosition(
+                                allOf(withId(cardId),
+                                        childAtPosition(
+                                                childAtPosition(
+                                                        allOf(withId(android.R.id.content),
+                                                                childAtPosition(
+                                                                        withId(R.id.action_bar_root),
+                                                                        1)),
+                                                        0),
+                                                position)),
+                                0),
+                        isDisplayed()));
+        appCompatImageButton.perform(click());
+
+    }
+
+    // click on scan button
+    public void step1(){
+        clickOnItem(R.id.scanButton, R.id.scanCard, 2);
+    }
+
+    // click on shrinker button
+    public void step2(){
+        clickOnItem(R.id.shrinkButton, R.id.shrinkerCard, 1);
+    }
+
+    // click on the health pack button
+    public void step3(){
+        clickOnItem(R.id.emsButton, R.id.emsCard, 4);
+    }
+
+    // click on the shield button
+    public void step4(){
+        clickOnItem(R.id.shieldButton, R.id.shieldCard, 3);
+    }
+
+    // click on "buy items"
+    public void step5(){
+        ViewInteraction appCompatButton3 = onView(
+                allOf(withId(R.id.BuyButton), withText("Buy Items"),
+                        childAtPosition(
+                                childAtPosition(
+                                        allOf(withId(android.R.id.content),
+                                                childAtPosition(
+                                                        allOf(withId(R.id.action_bar_root),
+                                                                childAtPosition(
+                                                                        withClassName(is("android.widget.FrameLayout")),
+                                                                        0)),
+                                                        1)),
+                                        0),
+                                5),
+                        isDisplayed()));
+        appCompatButton3.perform(click());
+    }
+
+    // check "Market" is displayed
+    public void step6(){
+        ViewInteraction textView = onView(withId(R.id.marketLabel));
+        textView.check(matches(withText("MARKET")));
+    }
+
+    @Test
+    public void newMarketActivityTest() throws InterruptedException {
+        // wait a bit for MarketActivity to be intended
+        Thread.sleep(5000);
```

Comment:
In general, it's not a good practice to use Thread.sleep(). However, unless I am mistaken, given your implementation, I don't see any other way of doing this